src/Datadog.Trace.ClrProfiler.Managed/ScopeDBFactory.cs

```diff
@@ -0,0 +1,80 @@
+using System;
+using System.Data;
+using Datadog.Trace.ClrProfiler.Integrations.AdoNet;
+using Datadog.Trace.ExtensionMethods;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.ClrProfiler
+{
+    internal static class ScopeDBFactory<T>
+    {
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(ScopeDBFactory<T>));
+
+        private static readonly string _fullName;
+        private static readonly string _dbTypeName;
+        private static readonly string _operationName;
+
+        static ScopeDBFactory()
+        {
+            var type = typeof(T);
+            _fullName = type.FullName;
+            _dbTypeName = ScopeFactory.GetDbType(type.Name);
+            _operationName = $"{_dbTypeName}.query";
```

Comment:
Just to confirm (and for prosperity), this is the source of the speedup, correct? i.e. we're not calling `GetType()` and allocating these strings for every call to `CreateDbCommandScope()`?

Reply:
that's right! optimisation using generics static to pre-calculate const values for the same type.